src/game.cpp

```diff
@@ -5742,6 +5758,13 @@ int game::assign_faction_id()
     return ret;
 }
 
+long game::assign_creature_id()
+{
+    long ret = next_creature_id;
```

Comment:
Crap, I totally forgot successive games, it's totally reasonable for a player to keep playing successive games in the same world indefinitely, and continually explore different areas.  This needs to be an order of magnitude larger, e.g. a long long (minimum 64 bits).  This might be what you intended in the first place.